logstash-core/lib/logstash/agent.rb

```diff
@@ -106,6 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
+      LogStash::Instrument::Collector.instance.agent = self
```

Comment:
Do the collector really need to know about the agent? Would it be better to start the webserver and give it the reference to the current agent?